danger/src/ChangelogChecker.ts

```diff
@@ -0,0 +1,167 @@
+  const {
+    DEFAULT_CHANGELOG_ENTRY_KEY: defaultEntry,
+    ...suggestedEntries
+  } = pullRequestManager.parseChangelogSuggestionFromDescription();
+
+  return packageNames.map(packageName => {
+    const message = suggestedEntries[packageName]?.message ?? defaultEntry.message;
+    const type = suggestedEntries[packageName]?.type ?? defaultEntry.type;
+    return {
+      packageName,
+      message,
+      type,
+    };
+  });
+}
+
+async function runAddChangelogCommandAsync(
+  suggestedEntries: PackageChangelogEntry[]
+): Promise<FixedChangelogEntry[]> {
+  await Promise.all(
+    suggestedEntries.map(entry =>
+      spawnAsync('et', [
+        `add-changelog`,
+        `--package`,
+        entry.packageName,
+        `--entry`,
+        entry.message,
+        `--author`,
+        prAuthor,
+        `--type`,
+        entry.type,
+        `--pull-request`,
+        `${pr.number}`,
+      ])
+    )
+  );
+
+  return Promise.all(
+    suggestedEntries.map(async entry => {
+      const changelogPath = path.join(
+        getExpoRepositoryRootDir(),
+        getPackageChangelogRelativePath(entry.packageName)
+      );
+      return {
+        ...entry,
+        content: await getFileContentAsync(changelogPath),
+        diff: await getFileDiffAsync(changelogPath),
+      };
+    })
+  );
+}
+
+function generateReport(
+  missingEntries: Array<{ packageName: string; diff: string }>,
+  url?: string | null
+) {
+  const message = missingEntries
+    .map(
+      entry =>
+        `- <code>${danger.github.utils.fileLinks(
+          [getPackageChangelogRelativePath(entry.packageName)],
+          false
+        )}</code>`
+    )
+    .join('\n');
+
+  const diff = '```diff\n' + missingEntries.map(entry => entry.diff).join('\n') + '```\n';
+  const pr = url ? `#### or merge this pull request: ${url}` : '';
+  fail(`
+📋 **Missing Changelog**
+------
+🛠 Add missing entires to:
+${message}`);
+
+  markdown(
+    `
+### 🛠 Suggested fixes:
+
+<details>
+  <summary>📋 Missing changelog</summary>
+
+  #### Apply suggested changes:
+${diff}
+${pr} 
+</details>`
+  );
+}
+
+/**
+ * This function checks if the changelog was modified, doing the following steps:
+ * - get packages which were modified but don't have changes in `CHANGELOG.md`
+ * - parse PR body to get suggested entries for those packages
+ * - run `et add-changelog` for each package to apply the suggestion
+ * - create a new PR
+ * - add a comment to inform about missing changelog
+ * - fail CI job
+ */
+export async function checkChangelog(): Promise<void> {
+  const modifiedPackages = groupBy(
+    danger.git.modified_files.filter(file => file.startsWith('packages')),
+    file => file.split(path.sep)[1]
+  );
+
+  const packagesWithoutChangelog = Object.entries(modifiedPackages)
+    .filter(([packageName, files]) => !isChangelogModified(packageName, files))
+    .map(([packageName, _]) => packageName);
```

Comment:
```suggestion
    .map(([packageName]) => packageName);
```
Probably that is possible 🤔